data/abilities.ts

```diff
@@ -357,7 +357,7 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		},
 		onAnyModifySpD(spd, target, source, move) {
 			const abilityHolder = this.effectState.target;
-			if (abilityHolder === target) return;
+			if (abilityHolder === target || target.hasAbility('Beads of Ruin')) return;
```

Comment:
we can probably just shorten these to just the hasAbility check right? since that must return true on the ability holder

Reply:
That feels correct